### PR TITLE
Fix highlighting issues

### LINKED ISSUE #2133: MatchesHighlighter doensn't respect HTML
`net.sf.jabref.logic.search.MatchesHighlighter#highlightWordsWithHTML` doesn't respect existing HTML.

It's clearly visible in the preview panel.
- If you use the Preview style, have `br` in your entry, and search for it: 
  html_preview
- if you use CitationStyle and search for `<`
  html_citationstyle

### PR DESCRIPTION
**Pull request by shitikanth**
Fixes #2133.

Earlier, highlighting was done manually by adding tags to matches in the html text being sent to the pane. Instead, I have opted to make use of the highlighting features of the JEditorPane class. This gives the additional benefit that the highlight colour is changed automatically based on the system theme. 

Changes 
- Created JEditorPaneWithHighlighting class
- Removed MatchHighlighter class and its tests
- Removed highlighting related tests from LayoutEntryTest
- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [x] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

**Comment by stefan-kolb** [comment]
:checkstyleMain[ant:checkstyle] [ERROR] /home/travis/build/JabRef/jabref/src/main/java/net/sf/jabref/gui/worker/CitationStyleWorker.java:6: 'javax.swing.JEditorPane' should be separated from previous imports. [ImportOrder]

**Comment by matthiasgeiger** [comment]
Thanks for your contribution!

What @stefan-kolb is telling you, that the tests are currently failing as you do not respect the order of the import we defined in your changed classes.

Please check https://github.com/JabRef/jabref/wiki/Guidelines-for-setting-up-a-local-workspace#get-an-ide-in-place and/or run `gradlew eclipse` to adjust the settings of IntelliJ/eclipse to order the imports correctly.

**Comment by shitikanth** [comment]
Thanks @stefan-kolb and @matthiasgeiger. I forgot to run the checks before submitting.

**Comment by stefan-kolb** [comment]
No worries @shitikanth 😄 These check style things are a little bit nitpicky 😉

File: src/main/java/net/sf/jabref/gui/PreviewPanel.java [hunk 1]
```diff
@@ -273,13 +272,19 @@ public void update() {
...
+            markHighlights();
         }
         else if (basePanel.isPresent()){
             citationStyleWorker = Optional.of(new CitationStyleWorker(this, previewPane));
             citationStyleWorker.get().execute();
         }
+
+    }
+
+    public void markHighlights() {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/PreviewPanel.java]: Does it make sense to set the highlight pattern automatically as a last step in `setPreviewLabel` ?

Comment (shitikanth) [inline src/main/java/net/sf/jabref/gui/PreviewPanel.java]: In some cases, we may want to set a preview label, but not want it to be search highlighted. 
For example, when CSL preview is being loaded or if the CSL preview generation failed and we are displaying an error message.

Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/PreviewPanel.java]: Ok, thanks for the follow-up. I merged your PR now.

**Pull request merged**